homeassistant/components/ecobee/climate.py

```diff
@@ -727,6 +755,63 @@ def resume_program(self, resume_all):
         )
         self.update_without_throttle = True
 
+    def set_sensors_used_in_climate(
+        self, sensors: list[str], climate_name: str | None = None
+    ) -> None:
+        """Set the sensors used on a climate for a thermostat."""
+        if climate_name is None:
+            climate_name = self.preset_mode
+
+        # Check if climate is an available preset option.
+        if climate_name not in self._preset_modes.values():
+            msg = f"Invalid climate name, available options are: {', '.join(self.preset_modes)}"
+            _LOGGER.error(msg)
+            raise HomeAssistantError(msg)
+
+        # Get device name from device id.
+        device_registry = dr.async_get(self.hass)
+        sensor_names: list[str] = []
+        for sensor in sensors:
+            sensor_registry = device_registry.async_get(sensor)
+            if sensor_registry and sensor_registry.name:
+                sensor_names.append(sensor_registry.name)
+
+        # Ensure sensors provided are available for thermostat.
+        if not set(sensor_names).issubset(set(self._sensors)):
+            msg = f"Invalid sensor for thermosat, available options are: {', '.join(self._sensors)}"
+            _LOGGER.error(msg)
+            raise HomeAssistantError(msg)
+
+        # Check if sensors are currently used on the climate for the thermostat.
+        current_sensors_in_climate = self.sensors_in_climate(climate_name)
+        if set(sensor_names) == set(current_sensors_in_climate):
+            msg = f"This action would not be an update, current sensors on climate ({climate_name}) are: {', '.join(current_sensors_in_climate)}"
+            _LOGGER.error(msg)
+            raise HomeAssistantError(msg)
+
+        _LOGGER.debug(
+            "Setting sensors %s to be used on thermostat %s for program %s",
+            sensor_names,
+            self.device_info.get("name"),
+            climate_name,
+        )
+        self.data.ecobee.update_climate_sensors(
+            self.thermostat_index, climate_name, sensor_names
+        )
+        self.update_without_throttle = True
+
+    def sensors_in_climate(self, climate_name: str | None = None) -> list[str]:
```

Comment:
```suggestion
    def _sensors_in_climate(self, climate_name: str) -> list[str]:
```

Please mark internally used functions as protected